src/CanCommunity.ts

```diff
@@ -149,52 +164,235 @@ export class CanCommunity {
+        community_account: communityAccount,
+        code_id: codeId,
+        right_accounts: input.voter_right_accounts || [],
+        right_pos_ids: input.voter_right_pos_ids || [],
+        is_amend_code: isAmendmentCode,
+      };
+      const packedParams = await serializeActionData(this.config, ActionNameEnum.SETVOTER, setVoterInput);
+      codeActions.push({
+        code_action: ActionNameEnum.SETVOTER,
+        packed_params: packedParams,
+      });
+    }
+
+    if (input.vote_duration > 0 && input.pass_rule > 0) {
+      const setVoteRuleInput: Setvoterule = {
+        community_account: communityAccount,
+        code_id: codeId,
+        pass_rule: input.pass_rule,
+        vote_duration: input.vote_duration,
+        is_amend_code: isAmendmentCode,
+      };
+      const packedParams = await serializeActionData(this.config, ActionNameEnum.SETVOTERULE, setVoteRuleInput);
+      codeActions.push({
+        code_action: ActionNameEnum.SETVOTERULE,
+        packed_params: packedParams,
+      });
+    }
 
-  async setRightHolderForCode(input: Setrightcode, execCodeInput?: ExecCodeInput): Promise<any> {
-    const packedParams = await serializeActionData(this.config, ActionNameEnum.SETRIGHTCODE, input);
+    if (input.exec_type > 0) {
+      const setExecTypeInput: Setexectype = {
+        community_account: communityAccount,
+        code_id: codeId,
+        exec_type: input.exec_type,
+        is_amend_code: isAmendmentCode,
+      };
+      const packedParams = await serializeActionData(this.config, ActionNameEnum.SETEXECTYPE, setExecTypeInput);
+      codeActions.push({
+        code_action: ActionNameEnum.SETEXECTYPE,
+        packed_params: packedParams,
+      });
+    }
 
-    return this.execCode(CODE_IDS.SET_RIGHT_HOLDER_FOR_CODE, ActionNameEnum.SETRIGHTCODE, packedParams, execCodeInput);
+    return codeActions;
   }
 
-  async setCollectionRuleForCode(input: Setcollectrl, execCodeInput?: ExecCodeInput): Promise<any> {
-    const packedParams = await serializeActionData(this.config, ActionNameEnum.SETCOLLECTRL, input);
+  async createCode(input: Createcode, execCodeInput?: ExecCodeInput): Promise<any> {
+    const packedParams = await serializeActionData(this.config, ActionNameEnum.CREATECODE, input);
 
-    return this.execCode(CODE_IDS.SET_COLLECTION_RULE_FOR_CODE, ActionNameEnum.SETCOLLECTRL, packedParams, execCodeInput);
+    const codeActions: ExecutionCodeData[] = [
+      {
+        code_action: ActionNameEnum.CREATECODE,
+        packed_params: packedParams,
+      },
+    ];
+
+    return this.execCode(CODE_IDS.CREATE_CODE, codeActions, CodeTypeEnum.NORMAL, execCodeInput);
   }
 
-  async setRightHolderForPosition(input: Setrightcode, execCodeInput?: ExecCodeInput): Promise<any> {
-    const packedParams = await serializeActionData(this.config, ActionNameEnum.SETRIGHTCODE, input);
+  async setRightHolderForCode(input: SetRightHolderInput, execCodeInput?: ExecCodeInput): Promise<any> {
```

Comment:
i think this func not only set right holder, it includes set type, right holder, rule... so i think should rename setRightHolderForCode to setExecutionRightForCode. @danielAlvess

Reply:
updated